### PR TITLE
Simplify gem versioning

### PR DESCRIPTION
**Pull request by carlosantoniodasilva**
This builds on top of https://github.com/judoscale/judoscale-ruby/pull/209, and is based on our discussion from a previous attempt in https://github.com/judoscale/judoscale-ruby/pull/208#discussion_r1589561120

We currently keep gems in sync and bump all of them together to improve the experience and facilitate maintenance. However that required manually syncing versions before a release, and we were looking for ways to automate that step. 

Instead, we're going to try eliminating the need to sync versions by having a single VERSION string, managed by judoscale-ruby, which gets automatically bumped by release-please for new releases. The gemspec and adapters were updated to refer to that single constant / source of truth, and gems will be built based on that, meaning bumping it will automatically work across all gems for new releases going forward.

It does make the fact that we report the same `adapter_version` across all adapters a bit unnecessary, but it's not much overhead and it's consistent with other adapters. The only drawback is in case we decide to circle back on the decision to use the same version across the board, but we'll cross that bridge if it gets here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -228,7 +228,6 @@ To install each gem onto your local machine, run `bundle exec rake install`.
 To release a new version:
 
 1. Use [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary), and release branches will be created automatically via [Release Please](https://github.com/google-github-actions/release-please-action). This updates the changelog and the version of judoscale-ruby.
-1. On the release branch, run `bin/sync-versions` to update the `VERSION` constants for each gem. Commit and push this change to the release branch.
```
Comment (carlosantoniodasilva) [inline README.md:231]: No need for syncing anymore.

File: judoscale-delayed_job/judoscale-delayed_job.gemspec [hunk 2]
```diff
@@ -1,10 +1,10 @@
 # frozen_string_literal: true
 
-require_relative "lib/judoscale/delayed_job/version"
+require_relative "../judoscale-ruby/lib/judoscale/version"
```
Comment (carlosantoniodasilva) [inline judoscale-delayed_job/judoscale-delayed_job.gemspec:3]: Require the relative file to refer to `Judoscale::VERSION` for gemspecs.

File: judoscale-delayed_job/lib/judoscale/delayed_job.rb [hunk 3]
```diff
@@ -2,13 +2,13 @@
 
 require "judoscale-ruby"
 require "judoscale/config"
-require "judoscale/delayed_job/version"
+require "judoscale/version"
```
Comment (carlosantoniodasilva) [inline judoscale-delayed_job/lib/judoscale/delayed_job.rb:5]: Eliminating the individual version files, refer/report the main VERSION across the board.

**Review by adamlogic — APPROVED** [review]
So much simpler. 💯

**Pull request merged**